java/client/src/main/java/glide/api/commands/ConnectionManagementClusterCommands.java

```diff
@@ -139,4 +139,42 @@ public interface ConnectionManagementClusterCommands {
      * }</pre>
      */
     CompletableFuture<ClusterValue<String>> clientGetName(Route route);
+
+    /**
+     * Echoes the provided <code>message</code> back.<br>
+     * The command will be routed a random node.
+     *
+     * @see <a href="https://redis.io/commands/echo/">redis.io</a> for details.
+     * @param message The message to be echoed back.
+     * @return The provided <code>message</code>.
+     * @example
+     *     <pre>{@code
+     * String payload = client.echo("GLIDE").get();
+     * assert payload.equals("GLIDE");
+     * }</pre>
+     */
+    CompletableFuture<String> echo(String message);
+
+    /**
+     * Echoes the provided <code>message</code> back.
+     *
+     * @see <a href="https://redis.io/commands/echo/">redis.io</a> for details.
+     * @param message The message to be echoed back.
+     * @param route Specifies the routing configuration for the command. The client will route the
+     *     command to the nodes defined by <code>route</code>.
+     * @return The provided <code>message</code>.
+     * @examples
+     *     <pre>{@code
+     * // Command sent to a single random node via RANDOM route, expecting a SingleValue result.
+     * String message = client.echo("GLIDE", RANDOM).get().getSingleValue();
+     * assert message.equals("GLIDE");
+     *
+     * // Command sent to all nodes via ALL_NODES route, expecting a MultiValue result.
+     * Map<String, String> msgForAllNodes = client.echo("GLIDE", ALL_NODES).get().getMultiValue();
+     * for(var msgPerNode : msgForAllNodes.entrySet()) {
```

Comment:
can also be done with:

`assert msgForAllNodes.get("node1.example.com:6379").equals("GLIDE");`

Reply:
Shachar asked us to not hardcode nodes in the examples and instead use for-loops.